### PR TITLE
[1464] Add course update

### PR DESCRIPTION
**Pull request by heidar**
### Context

Frontend sends us data to update a course (or the enrichment rather)

### Changes proposed in this pull request

Update endpoint for courses.

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by heidar** [comment]
[[M] Course update action in mc-be](https://trello.com/c/1lZpQcIX/1464-m-course-update-action-in-mc-be)

File: app/controllers/api/v2/courses_controller.rb [hunk 1]
```diff
@@ -52,6 +73,27 @@ def build_course
...
+          .except(:id, :type)
+          .permit(
+            :about_course,
+            :course_length,
+            :fee_details,
+            :fee_international,
+            :fee_uk_eu,
+            :financial_support,
+            :how_school_placements_work,
+            :internal_server_error,
```
Comment (defong) [inline app/controllers/api/v2/courses_controller.rb]: internal_server_error
?

permit as a a param?

Comment (misaka) [reply app/controllers/api/v2/courses_controller.rb]: Ya, wat? I don't know how that got in there, will double-check and remove. Thanks!

File: app/deserializers/api/v2/deserializable_course.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                 :fee_international,
+                 :fee_uk_eu,
+                 :financial_support,
+                 :how_school_placements_work,
+                 :interview_process,
+                 :other_requirements,
+                 :personal_qualities,
+                 :qualifications,
+                 :salary_details,
+                 :findable?,
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from the site statuses associated with the course. Should be removed.

File: app/deserializers/api/v2/deserializable_course.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                 :open_for_applications?,
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from the site statuses associated with the course. Should be removed.

File: app/deserializers/api/v2/deserializable_course.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                 :has_vacancies?,
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from the site statuses associated with the course. Should be removed.

File: app/deserializers/api/v2/deserializable_course.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                 :course_code,
+                 :name,
+                 :study_mode,
+                 :description,
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from the course info. Should be removed.

File: app/deserializers/api/v2/deserializable_course.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+                 :content_status,
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from the course enrichment. Should be removed.

File: app/deserializers/api/v2/deserializable_course.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+                 :ucas_status,
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from the course info. Should be removed.

File: app/deserializers/api/v2/deserializable_course.rb [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+                 :funding,
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from the course info. Should be removed.

File: app/deserializers/api/v2/deserializable_course.rb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+                 :applications_open_from,
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from the course info. Should be removed.

File: app/deserializers/api/v2/deserializable_course.rb [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+                 :level,
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from subject info. Should be removed.

File: app/deserializers/api/v2/deserializable_course.rb [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+                 :is_send?
```
Comment (misaka) [inline app/deserializers/api/v2/deserializable_course.rb]: There's no way to set this value, it's derived from subject info. Should be removed.

File: app/controllers/api/v2/courses_controller.rb (hunk 1 above)
Comment (timabell) [reply app/controllers/api/v2/courses_controller.rb]: done

**Comment by timabell** [comment]
rebased & comments addressed

**Pull request merged**